src/clients/Client.cc

```diff
@@ -147,6 +147,9 @@ Client::setFinalReply(HttpReply *rep)
     // give entry the reply because haveParsedReplyHeaders() expects it there
     entry->replaceHttpReply(theFinalReply, false); // but do not write yet
     haveParsedReplyHeaders(); // update the entry/reply (e.g., set timestamps)
+    if (finalReplyIsFresh())
+        entry->clearPublicKeyScope();
+
     if (!EBIT_TEST(entry->flags, RELEASE_REQUEST) && blockCaching())
         entry->release();
```

Comment:
Calling `release()` may make the entry private. To avoid expensive operations, all other factors being equal, it would be better to handle ksRevalidation scope (if it is still there) _after_ this call.